lib/Conversion/AIEVecToLLVM/AIEVecToLLVM.cpp

```diff
@@ -117,6 +117,42 @@ struct BufferParams {
   uint32_t square;
 };
 
+static VectorType getFlattenedVectorType(VectorType vecTy) {
+  if (vecTy.getRank() == 1)
+    return vecTy;
+  auto shape = vecTy.getShape();
+  return VectorType::get(
+      {std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<>())},
+      vecTy.getElementType());
+}
+
+// sgn_x: Sign mask of matrix X. If it is one matrix X is interpreted as
+// signed, else it treated as unsigned.
+// sgn_y: Sign mask of matrix Y. If it is one matrix Y is interpreted as
+// signed, else it treated as unsigned.
+// amode/bmode/variant: config acc width, mul precision, and mul mode
+// zero_acc: Zeroing of acc1. If it is one then acc1 is zeroed.
+// shift16: Shift mask of acc1. If a bit is set the <<16 operation will be
+// executed on acc1.
+// sub_mul: Negation mask of the matrix multiplication result. If it is
+// one the result of the operation will be negated.
+// sub_acc1: Negation mask of acc1. If it is one acc1 will be negated.
+// sub_acc2: Negation mask of acc2. If it is one acc2 will be negated.
+// sub_mask: Negation mask of complex multiplications. Negates a term of a
+// complex multiplication.
+static inline int aiev2_vmac_compute_control(int sgn_x, int sgn_y, int amode,
```

Comment:
We can leverage this helper in the conversion of `aievec::MatMulOp` as well.

Reply:
In a different patch, yes, it's in my list.